_posts/2015-07-20-CocoaPods-0.38.markdown

```diff
@@ -0,0 +1,121 @@
+
+<!-- more -->
+
+CocoaPods 0.38 has been released.
+This is the first update since WWDC and it brings support for Apple's latest releases including Xcode 7 and watchOS.
+It also includes lots of improvements that make building Pods easier.
+Furthermore it rolls out [Stats](http://blog.cocoapods.org/Stats/) for everyone.
+
+
+### watchOS
+
+The update allows you to use Pods on watchOS.
+As Pod Author, you have to specify the [deployment target
+for watchOS](https://guides.cocoapods.org/syntax/podspec.html#watchos)
+ in the Podspec, as seen below:
+
+```ruby
+Pod::Spec.new do |s|
+  # …
+  s.watchos.deployment_target = '2.0'
+end
+```
+ 
+This also means that all Pods, which should be available on watchOS must be explicitly updated therefore.
+
+
+### Target Deduplication
+
+CocoaPods de-duplicates now Pod Targets, which means that it recognizes when a dependency is used multiple times across different user targets, but needs to be built only once. But there can be some reasons, when this is not possible and they still need to be generated in different variants and so also build multiple times. The targets in `Pods.xcodeproj` need to be duplicated when one of the following condition applies:
+
+* They are used on different platforms.
+* They are used with differents sets of subspecs.
+* They have any dependency which needs to be duplicated.
+
+This fixes several issues with Archiving apps with extensions, but also reduces build times significantly.
+
+
+### Removal of the Enviroment Header
+
+After updating and running `pod install` on your integration, you will see that some header files in the Pods directory won't be generated anymore matching the pattern `Pods-*-environment.h`.
+This was the environment header, which contained precompiler macro definitions with the version components of each dependency used in the integrated target.
+It was visible not only for the user target, but also for all its dependencies.
+It could be used to make additional features available if a given pod is present.
+But this was also the cause of some issues.
+One of them was that it wouldn't be possible to de-duplicate frameworks.
+If you relied on that, you can still use the Clang file checking macro [`__has_include`](http://clang.llvm.org/docs/LanguageExtensions.html#include-file-checking-macros) to determine by the presence of a header, whether a specific dependency is available or not:
+
+```objective-c
+#if __has_include("QueryKit.h")
+  #import <QueryKit/QueryKit.h>
+#endif
+```
+
+To make optional behavior in your Pod available, you should use [subspecs](https://guides.cocoapods.org/syntax/podspec.html#group_subspecs), exclude the subspec from the `default_subspecs` and declare there an explicit dependency, which is versioned and will affect the dependency resolution only when the user explicitly includes the subspec.
+
+
+### Breaking Change to the Hooks-API
```

Comment:
Heard from a user on Twitter that it poses a problem for them, because they're using multiple versions of CP on CI for the same project. We could mention how one can have version-specific code in the Podfile, using a snippet like this <https://gist.github.com/neonichu/0afaee43bcb3c90bcc23>